src/util/include/util/DispatchPool.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Xournal++
+ *
+ * Template class for a dispatch pool and listeners
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <cassert>
+#include <memory>
+#include <vector>
+
+namespace xoj::util {
+
+/**
+ * @brief Generic template for dispatchers.
+ *
+ * The listeners must implement functions
+ *      void on(Arg&&...);
+ * that the dispatcher will call through dispatch().
+ */
+template <class ListenerT>
+class DispatchPool final {
+public:
+    using listener_type = ListenerT;
+
+    template <typename... Args>
```

Comment:
Needs a doc comment.
```suggestion
    /**
     * @brief Invokes the `on()` method of all registered `ListenerT`s.
     */
    template <typename... Args>
```